sympy/simplify/simplify.py

```diff
@@ -2540,8 +2541,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or isinstance(expr, MatrixSymbol) or (
+            expr.is_Atom or expr in (exp_polar(0), exp_polar(1)))):
```

Comment:
This is an interesting one. At first, without this change, ```simplify``` will not work with MatrixElement expression
```
>>> M = MatrixSymbol('M', 10, 10)
>>> e = M[0, 0] + M[5, 4] + 1304
>>> result = simplify(e)
>>> result
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/core/basic.py", line 392, in __repr__
    return sstr(self, order=None)
  File "sympy/printing/str.py", line 742, in sstr
    s = p.doprint(expr)
  File "sympy/printing/printer.py", line 233, in doprint
    return self._str(self._print(expr))
  File "sympy/printing/printer.py", line 257, in _print
    return getattr(self, printmethod)(expr, *args, **kwargs)
  File "sympy/printing/str.py", line 56, in _print_Add
    t = self._print(term)
  File "sympy/printing/printer.py", line 257, in _print
    return getattr(self, printmethod)(expr, *args, **kwargs)
  File "sympy/printing/str.py", line 227, in _print_MatrixElement
    return self._print(expr.parent) + '[%s, %s]'%(expr.i, expr.j)
TypeError: unsupported operand type(s) for +: 'Symbol' and 'str'
```
The strange thing is that ```simplify``` calls succeeds. But the returned result is not printable. I managed to trace the problem to this ```powsimp``` function